client/src/components/itineraries/Itineraries.js

```diff
@@ -0,0 +1,97 @@
+import React, { Component } from "react";
+import ItinerariesList from "./ItinerariesList";
+import Footer from "../Footer";
+import { connect } from "react-redux";
+import { fetchItineraries } from "../../store/actions/itinerariesActions";
+import { fetchActivities } from "../../store/actions/activitiesActions";
+import {
+  fetchComments,
+  postComment,
+  deleteComment
+} from "../../store/actions/commentsActions";
+import M from "materialize-css";
+
+export class Itineraries extends Component {
+  async componentDidMount() {
+    await this.props.fetchItineraries(this.props.match.params.id);
+    await this.props.fetchActivities(this.props.match.params.id);
+    await this.props.fetchComments(this.props.match.params.id);
+  }
+
+  componentDidUpdate() {
+    var elems = document.querySelectorAll(".collapsible");
+    M.Collapsible.init(elems, { inDuration: 300 });
+  }
+
+  loader() {
+    return (
+      <div>
```

Comment:
You might want to make this a re-usable component since Ive seen it a couple times now.